test/OpenSpaceToolkit/Astrodynamics/Trajectory/Validation/Propagator.cross.validation.cpp

```diff
@@ -0,0 +1,1584 @@
+
+            referencePositionArrayGCRF.add({
+                referenceRow[1].accessReal(),
+                referenceRow[2].accessReal(),
+                referenceRow[3].accessReal(),
+            });
+            referenceVelocityArrayGCRF.add({
+                referenceRow[4].accessReal(),
+                referenceRow[5].accessReal(),
+                referenceRow[6].accessReal(),
+            });
+
+            referencePositionArray_ITRF.add({
+                referenceRow[7].accessReal(),
+                referenceRow[8].accessReal(),
+                referenceRow[9].accessReal(),
+            });
+            referenceVelocityArray_ITRF.add({
+                referenceRow[10].accessReal(),
+                referenceRow[11].accessReal(),
+                referenceRow[12].accessReal(),
+            });
+        }
+
+        const Shared<const Frame> itrfSPtr = Frame::ITRF();
+
+        // Current state and instant setup
+        const Instant startInstant = instantArray[0];
+
+        // Setup initial conditions
+        const State state = {
+            startInstant,
+            Position::Meters({referencePositionArrayGCRF[0]}, gcrfSPtr_),
+            Velocity::MetersPerSecond({referenceVelocityArrayGCRF[0]}, gcrfSPtr_),
+        };
+
+        // Propagate all states
+        const Array<State> propagatedStateArray = defaultPropagator_.calculateStatesAt(state, instantArray);
+
+        // Validation loop
+        for (size_t i = 0; i < instantArray.getSize(); i++)
+        {
+            // GCRF Compare
+            const Position positionGCRF = propagatedStateArray[i].getPosition();
+            const Velocity velocityGCRF = propagatedStateArray[i].getVelocity();
+
+            const double positionErrorGCRF = (positionGCRF.accessCoordinates() - referencePositionArrayGCRF[i]).norm();
+            const double velocityErrorGCRF = (velocityGCRF.accessCoordinates() - referenceVelocityArrayGCRF[i]).norm();
+
+            ASSERT_EQ(*Frame::GCRF(), *positionGCRF.accessFrame());
+            ASSERT_EQ(*Frame::GCRF(), *velocityGCRF.accessFrame());
+
+            ASSERT_GT(2e-7, positionErrorGCRF);
+            ASSERT_GT(2e-10, velocityErrorGCRF);
+
+            // ITRF Compare
+            const Position position_ITRF = (propagatedStateArray[i].inFrame(itrfSPtr)).getPosition();
+            const Velocity velocity_ITRF = (propagatedStateArray[i].inFrame(itrfSPtr)).getVelocity();
+
+            const double positionError_ITRF =
+                (position_ITRF.accessCoordinates() - referencePositionArray_ITRF[i]).norm();
+            const double velocityError_ITRF =
+                (velocity_ITRF.accessCoordinates() - referenceVelocityArray_ITRF[i]).norm();
+
+            ASSERT_EQ(*Frame::ITRF(), *position_ITRF.accessFrame());
+            ASSERT_EQ(*Frame::ITRF(), *velocity_ITRF.accessFrame());
+
+            ASSERT_GT(2e-2, positionError_ITRF);
+            ASSERT_GT(2e-5, velocityError_ITRF);
+
+            // Results console output
```

Comment:
Leftover?

Reply:
No these are kept everywhere for validation tests in ostk-astro so that if something fails its easier to debug